phoenix-tools/src/main/java/org/apache/phoenix/schema/SchemaExtractionProcessor.java

```diff
@@ -0,0 +1,410 @@
+
+        populateDefaultProperties(table);
+        setPTableProperties(table);
+        setHTableProperties(htd);
+        setHColumnFamilyProperties(hcd);
+
+        String columnInfoString = getColumnInfoStringForTable(table);
+        String propertiesString = convertPropertiesToString();
+
+        return generateTableDDLString(columnInfoString, propertiesString, pSchemaName, pTableName);
+    }
+    private String generateTableDDLString(String columnInfoString, String propertiesString,
+            String pSchemaName, String pTableName) {
+        String pTableFullName = SchemaUtil.getQualifiedTableName(pSchemaName, pTableName);
+        StringBuilder outputBuilder = new StringBuilder(String.format(CREATE_TABLE, pTableFullName));
+        outputBuilder.append(columnInfoString).append(" ").append(propertiesString);
+        return outputBuilder.toString();
+    }
+
+    private void populateDefaultProperties(PTable table) {
+        Map<String, String> propsMap = HColumnDescriptor.getDefaultValues();
+        for (Map.Entry<String, String> entry : propsMap.entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            defaultProps.put(key, value);
+            if(key.equalsIgnoreCase(HColumnDescriptor.BLOOMFILTER) || key.equalsIgnoreCase(
+                    HColumnDescriptor.COMPRESSION)) {
+                defaultProps.put(key, "NONE");
+            }
+            if(key.equalsIgnoreCase(HColumnDescriptor.DATA_BLOCK_ENCODING)) {
+                defaultProps.put(key, String.valueOf(SchemaUtil.DEFAULT_DATA_BLOCK_ENCODING));
+            }
+        }
+        defaultProps.putAll(table.getDefaultPropertyValues());
+    }
+
+    private void setHTableProperties(HTableDescriptor htd) {
+        Map<ImmutableBytesWritable, ImmutableBytesWritable> propsMap = htd.getValues();
+        for (Map.Entry<ImmutableBytesWritable, ImmutableBytesWritable> entry : propsMap.entrySet()) {
+            ImmutableBytesWritable key = entry.getKey();
+            ImmutableBytesWritable value = entry.getValue();
+            if(Bytes.toString(key.get()).contains("coprocessor") || Bytes.toString(key.get()).contains(
+                    HTableDescriptor.IS_META)) {
+                continue;
+            }
+            defaultProps.put(Bytes.toString(key.get()), "false");
+            definedProps.put(Bytes.toString(key.get()), Bytes.toString(value.get()));
+        }
+    }
+
+    private void setHColumnFamilyProperties(HColumnDescriptor columnDescriptor) {
+        Map<ImmutableBytesWritable, ImmutableBytesWritable> propsMap = columnDescriptor.getValues();
+        for (Map.Entry<ImmutableBytesWritable, ImmutableBytesWritable> entry : propsMap.entrySet()) {
+            ImmutableBytesWritable key = entry.getKey();
+            ImmutableBytesWritable value = entry.getValue();
+            definedProps.put(Bytes.toString(key.get()), Bytes.toString(value.get()));
```

Comment:
This will be incorrect for multiple column families. (e.g two CFs with different values for the same property will overwrite each other)

Reply:
Would like to get it fixed in subsequent commits.

Reply:
Added a message to throw Exception with "Not Supported" message